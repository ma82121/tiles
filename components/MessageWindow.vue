<template>
  <div v-if="isShow" class="message-window">
    {{ displayMessage }}
  </div>
</template>

<script>
export default {
  name: 'MessageWindow',
  props: {
    isShow: {
      type: Boolean,
      default: true
    },
    message: {
      type: String,
      default: ''
    }
  },
  data() {
    return {
      timer: null,
      speed: 30,
      displayMessage: '',
      count: 0
    }
  },
  watch: {
    message() {
      this.play()
    }
  },
  mounted() {
    this.play()
  },
  destroyed() {
    clearInterval(this.timer)
  },
  methods: {
    update() {
      if (this.count > this.message.length) {
        this.stop()
      } else {
        const text = this.message.substr(0, this.count)
        this.displayMessage = text
        this.count++
      }
    },
    play() {
      this.reset()
      this.timer = setInterval(this.update, 1000 / this.speed)
    },
    stop() {
      this.displayMessage = this.message
      clearInterval(this.timer)
    },
    reset() {
      this.timer = null
      this.displayMessage = ''
      this.count = 0
    }
  }
}
</script>

<style lang="scss" scoped>
.message-window {
  position: relative;
  margin: 0 auto;
  padding: 2px;
  width: 300px;
  height: 60px;
  box-sizing: border-box;
  font-size: 0.9rem;
  font-weight: bold;
}
</style>
