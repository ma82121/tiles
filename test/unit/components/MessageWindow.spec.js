import { shallowMount } from '@vue/test-utils'
import MessageWindow from '@/components/MessageWindow'

describe('MessageWindow', () => {
  test('snapshot', () => {
    const wrapper = shallowMount(MessageWindow)
    expect(wrapper.element).toMatchSnapshot()
  })
})
