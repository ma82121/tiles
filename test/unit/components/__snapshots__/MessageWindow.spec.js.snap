// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageWindow snapshot 1`] = `
<div
  class="message-window"
>
  
  

</div>
`;
